[10.0] [MIG] product_margin_classification

FFernandez-PlanetaTIC opened the pull request.

legalsylvain committed:
[ADD] new module 'product_margin_classification';

legalsylvain committed:
[FIX] typo;
[ADD] fr translation;
[FIX] OCA convention;

legalsylvain committed:
[FIX] OCA as author

legalsylvain committed:
[IMP] add need action to show product with prices different from the theoritical one

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

legalsylvain committed:
[FIX+IMP] product_margin_classification
[FIX] return correct current company when creating a new margin classification
[FIX] incorrect 'type' account.tax field
[FIX] Fix demo data. (ref instead of eval)
[FIX] replace smart buttons on product view by regular buttons according to smart buttons implementation.
[IMP] display buttons only if required on margin classification form
[IMP] Add test file
[IMP] replace integrated tree view in margin classification form by a smart button and a dedicated tree view
[ADD] complete margin concept with two fields 'Markup' and 'Profit Margin'

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update product_margin_classification.pot

legalsylvain committed:
[FIX] wording and update translation

oca-travis committed:
[UPD] Update product_margin_classification.pot

OCA-git-bot committed:
product_margin_classification 8.0.2.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-8.0/sale-workflow-8.0-product_margin_classification
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-8-0/sale-workflow-8-0-product_margin_classification/

FFernandez-PlanetaTIC commented:
@MPoch-PlanetaTIC please:

- Fix version
- Remove PlanetaTIC from authors
- Add yourself to contributors
- Fix website to https://github.com/OCA/sale-workflow

Thanks!

legalsylvain reviewed (changes requested):
hi @FFernandez-PlanetaTIC. Thanks a lot for porting this module. 

When you migrate an OCA module, please follow the OCA guidelines, to keep github history. (Here, you port from 8.0 to 10.0) https://github.com/OCA/sale-workflow/tree/8.0/product_margin_classification

You can follow this procedure here (replacing 9.0 by 8.0) : 
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0#technical-method-to-migrate-a-module-from-90-to-100-branch

kind regards.

FFernandez-PlanetaTIC commented:
Thanks @legalsylvain , we will do it following the OCA guidelines

MPoch-PlanetaTIC committed:
[MIG] product_margin_classification: Migration to 10.0

legalsylvain commented on product_margin_classification/views/menu.xml:
```diff
@@ -4,11 +4,11 @@ Copyright (C) 2016-Today GRAP (http://www.grap.coop)
 
     <menuitem id="menu_product_margin_classification"
-        parent="base.menu_product"
+        parent="sale.menu_product"
```
you depend on sale, but the module is not in the dependencies, in the manifest.

legalsylvain commented on product_margin_classification/security/ir.model.access.csv:
```diff
@@ -1,3 +1,3 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 product_margin_classification_user,product_margin_classification user,model_product_margin_classification,base.group_user,1,0,0,0
-product_margin_classification_manager,product_margin_classification manager,model_product_margin_classification,base.group_sale_manager,1,1,1,1
+product_margin_classification_manager,product_margin_classification manager,model_product_margin_classification,sales_team.group_sale_manager,1,1,1,1
```
same here, you depends on sales_team but it is not in the manifest file.

legalsylvain commented on product_margin_classification/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,35 @@
+displayed, based on the Margin Classification and a button is available to
+change sale price.
+
+.. image:: /product_margin_classification/static/description/product_template_form.png
```
could your replace by : 

`` .. image:: ../static/description/product_template_form.png``

(3 occurencies)

It will work in all cases. (apps store, github, etc...)

FFernandez-PlanetaTIC replied:
@MPoch-PlanetaTIC please, replace the other 2 occurencies, thanks

legalsylvain commented on product_margin_classification/models/product_template.py:
```diff
@@ -49,15 +49,16 @@ def _compute_theoretical_multi(self):
                 multi = 1 + classification.markup
-                if template.taxes_id.filtered(lambda x: x.type != 'percent'):
+                if template.taxes_id.filtered(
+                        lambda x: x.amount_type != 'percent'):
```
not a blocking point, but in OCA convention, we avoid to make such changes that are cosmetic. (no reason to write on two lines).

MPoch-PlanetaTIC replied:
Line 52 was longer than 80 characters. So I split it.

legalsylvain replied:
ok. nice so.

legalsylvain commented on product_margin_classification/__manifest__.py:
```diff
@@ -5,11 +5,26 @@
-    'website': 'http://www.grap.coop',
+    'website': 'https://github.com/OCA/sale-workflow',
     'license': 'AGPL-3',
+    'description': """
```
this 'description' is useless, as there is the readme folder. when merging your PR, the readme.rst file will be generated by bot.

legalsylvain reviewed (changes requested):
Hi @FFernandez-PlanetaTIC. 
some minors remarks. 
2 problems about dependencies that should be fixed. (i guess that setting dependencies to sales_team will fix both problem).

Otherwise, LGTM. Thanks a lot for this work.

MPoch-PlanetaTIC commented:
Thank you @legalsylvain for your support in my first migration.

FFernandez-PlanetaTIC commented on product_margin_classification/models/product_margin_classification.py:
```diff
@@ -0,0 +1,163 @@
+
+    price_round = fields.Float(
+        string='Price Rounding',
+        digits_compute=dp.get_precision('Product Price'),
```
parameter 'digits_compute' is no longer supported; use 'digits' instead

FFernandez-PlanetaTIC commented on product_margin_classification/models/product_margin_classification.py:
```diff
@@ -0,0 +1,163 @@
+
+    price_surcharge = fields.Float(
+        string='Price Surcharge',
+        digits_compute=dp.get_precision('Product Price'),
```
parameter 'digits_compute' is no longer supported; use 'digits' instead

FFernandez-PlanetaTIC commented:
@MPoch-PlanetaTIC please, could you review test `test_04_sale_order_triple_discount_invoicing` in module `sale_triple_discount`? thanks

FFernandez-PlanetaTIC commented:
Hi,

`product_special_type` is not installable but it raises this error on test:

https://travis-ci.org/OCA/sale-workflow/jobs/566429674#L2027

> 2019-08-01 11:45:43,770 6757 INFO ? werkzeug: 127.0.0.1 - - [01/Aug/2019 11:45:43] "GET /web/content/557-b9cb9dc/report.assets_common.1.css HTTP/1.1" 500 -
> 2019-08-01 11:45:43,775 6757 ERROR ? werkzeug: Error on request:
> Traceback (most recent call last):
> File "/home/travis/virtualenv/python2.7_with_system_site_packages/lib/python2.7/site-packages/werkzeug/serving.py", line 193, in run_wsgi
> execute(self.server.app)
> File "/home/travis/virtualenv/python2.7_with_system_site_packages/lib/python2.7/site-packages/werkzeug/serving.py", line 181, in execute
> application_iter = app(environ, start_response)
> File "/home/travis/odoo-10.0/odoo/service/server.py", line 307, in app
> return self.app(e, s)
> File "/home/travis/odoo-10.0/odoo/service/wsgi_server.py", line 186, in application
> return application_unproxied(environ, start_response)
> File "/home/travis/odoo-10.0/odoo/service/wsgi_server.py", line 172, in application_unproxied
> result = handler(environ, start_response)
> File "/home/travis/odoo-10.0/odoo/http.py", line 1325, in __call__
> self.load_addons()
> File "/home/travis/odoo-10.0/odoo/http.py", line 1346, in load_addons
> m = __import__('odoo.addons.' + module)
> File "/home/travis/odoo-10.0/odoo/modules/module.py", line 81, in load_module
> execfile(modfile, new_mod.__dict__)
> File "/home/travis/build/OCA/sale-workflow/product_special_type/__init__.py", line 22, in <module>
> import product
> File "/home/travis/build/OCA/sale-workflow/product_special_type/product.py", line 22, in <module>
> from osv import osv, fields
> ImportError: No module named osv

Can someone give us a hint?

Thanks!

pedrobaeza commented:
Rebased after https://github.com/OCA/sale-workflow/commit/8428446c79981aee281eb1773ff84c5a32d02067.

This is why in v11 we switch the approach to cleaning totally the branch.

MPoch-PlanetaTIC committed:
[FIX] product_margin_classification: Migration to 10.0

legalsylvain commented:
Hi @FFernandez-PlanetaTIC. Thanks for all the changes. Code review OK. I just tested the module on runbot, and it seems that in V10 percent field are now * 100. 
I mean in V8 : 10% tax value was : ``0.1``, in V10 tax value is ``10.0``

For consistency reason, I think that OCA modules should adopt the same logic. It requires : 
- change the field and the display ;
- make a migration script for people that comes from V8. (my case).

I'll do a PR against your PR, to apply such changes. What do you think ?

regards.

FFernandez-PlanetaTIC commented:
Hi @legalsylvain , that would be fine, thanks!

legalsylvain committed:
[IMP] Migration to V10.0
- remove obsolete oldname fields
- harmonize field name
- rename field selection values
- Multiply margin and markup field by 100, according to new convention of Odoo, regarding Float 'Percent' field. (Cf VAT)
- use new font awsome icons

legalsylvain commented:
Done here, with various minor changes.
https://github.com/PlanetaTIC/sale-workflow/pull/1
Could you review, and if all is ok, merge this sub PR in this one ? 

kind regards.

legalsylvain reviewed (approved):
- No need to include travis fix (postgtresql into this PR). 

otherwise, LGTM. Thanks ! 

Note : i think that for next version, we should move this module into this repository https://github.com/OCA/margin-analysis .

what do you think ?

MPoch-PlanetaTIC committed:
[FIX] product_margin_classification: :green_apple: tests

FFernandez-PlanetaTIC commented:
Moving this module makes sense

FFernandez-PlanetaTIC commented:
Fails in `sale_triple_discount` tests will be fixed in #915

FFernandez-PlanetaTIC commented:
Ready for review

FFernandez-PlanetaTIC committed:
[FIX] product_margin_classification: perf issue on change theor diff

Fix performance issue on change theoretical difference when many
products in margin classification

legalsylvain commented:
/ocabot merge

legalsylvain commented:
FYI, I'm working on the migration in V12. I'll ping you when the PR is available in margin-analysis repository.

kind regards.
